docs/development/design/organizations.rst

```diff
@@ -0,0 +1,145 @@
+
+We have the following features in the commercial site that we don't have on the community site:
+
+- Owners
+- Teams
+- Permissions
+- Subscriptions
+
+Owners should be included to represent owners of the current organization.
+
+Teams, this is also handy to manage access to different projects under the same organization.
+
+Permissions,
+currently we have two type of permissions for teams: admin and read only.
+Read only permissions doesn't make sense in the community site since we only support public projects/versions
+(we do support private versions now, but we are planning to remove those).
+So, we should only support admin permissions for teams.
+
+Subscriptions, this is only valid for the corporate site,
+since we don't charge for use in the community site.
+
+How to migrate current projects
+-------------------------------
+
+Since we are not replacing the current implementation,
+we don't need to migrate current projects from the community site nor from the corporate site.
+
+How to migrate the organizations app
+------------------------------------
+
+The migration can be split in:
+
+#. Remove/simplify code from the organizations app on the corporate site.
+#. Isolate/separate models and code that isn't going to be moved.
+#. Start by moving the models, managers, and figure out how to handle migrations.
+#. Move the rest of the code as needed.
+#. Activate organizations app on the community site.
+#. Integrate the code from the community site to the new code.
+
+We should start by removing unused features and dead code from the organizations in the corporate site,
+and simplify existing code if possible (some of this was already done).
+
+Isolate/separate the models to be moved from the ones that aren't going to be moved.
+We should move the models that aren't going to me moved to another app.
+
+- Plan
+- PlanFeature
+- Subscription
+
+This app can be named *subscriptions*.
+We can get around the table names and migrations by setting the explicitly the table name to `organizations_<model>`,
+and doing a fake migration.
+Following suggestions in https://stackoverflow.com/questions/48860227/moving-multiple-models-from-one-django-app-to-another.
+Code related to subscriptions should be moved out from the organizations app.
+
+After that, it should be easier to move the organizations *app* (or part of it)
+to the community site (and no changes to table names would be required).
+
+We start by moving the models.
+
+- Organization
+- OrganizationOwner
+- Team
+- TeamInvite
+- TeamMember
+
+Migrations aren't moved, since all current migrations depend on other models that aren't
+going to be moved.
+In the community site we run an initial migration,
+for the corporate site we run a fake migration.
+
+For managers and querysets that depend on subscriptions,
+we can use our pattern to make overridable classes (inheriting from ``SettingsOverrideObject``).
+
+Templates, urls, views, forms, notifications, signals, tasks can be moved later
```

Comment:
I think it would be useful to have a bit more of this fleshed out. I think that moving the models is definitely the best first step, but it won't do us much good without migrating the queryset permission logic as well.